tests/h/auth/tokens_test.py

```diff
@@ -1,85 +1,48 @@
+from pytest import param
 
 from h.auth import tokens
 
 
-class TestToken:
-    def test_token_with_no_expiry_is_valid(self):
-        token = tokens.Token(mock.Mock(expires=None, userid="acct:foo@example.com"))
-
-        assert token.is_valid()
-
-    def test_token_with_future_expiry_is_valid(self):
-        token = tokens.Token(
-            mock.Mock(userid="acct:foo@example.com", expires=_seconds_from_now(1800))
-        )
+def _seconds_from_now(seconds):
+    return datetime.datetime.utcnow() + datetime.timedelta(seconds=seconds)
 
-        assert token.is_valid()
 
-    def test_token_with_past_expiry_is_not_valid(self):
+class TestToken:
+    @pytest.mark.parametrize(
+        "expires,is_valid",
+        (
+            param(None, True, id="no expiry"),
+            param(_seconds_from_now(1800), True, id="future expiry"),
+            param(_seconds_from_now(-1800), False, id="past expiry"),
+        ),
+    )
+    def test_it(self, expires, is_valid, factories):
         token = tokens.Token(
-            mock.Mock(userid="acct:foo@example.com", expires=_seconds_from_now(-1800))
+            factories.OAuth2Token(userid="acct:foo@example.com", expires=expires)
         )
 
-        assert not token.is_valid()
-
-
-VALID_TOKEN_EXAMPLES = [
-    # Valid
-    lambda k: jwt.encode({"exp": _seconds_from_now(3600)}, key=k),
-    # Expired, but within leeway
-    lambda k: jwt.encode({"exp": _seconds_from_now(-120)}, key=k),
-]
-
-INVALID_TOKEN_EXAMPLES = [
-    # Expired 1 hour ago
-    lambda k: jwt.encode({"exp": _seconds_from_now(-3600)}, key=k),
-    # Incorrect encoding key
-    lambda k: jwt.encode({"exp": _seconds_from_now(3600)}, key="somethingelse"),
-]
+        assert token.is_valid() == is_valid
 
 
 class TestAuthToken:
-    def test_retrieves_token_for_request(self, pyramid_request):
-        pyramid_request.headers["Authorization"] = "Bearer abcdef123"
-
-        result = tokens.auth_token(pyramid_request)
-
-        assert result == "abcdef123"
-
-    def test_returns_none_when_no_authz_header(self, pyramid_request):
-        result = tokens.auth_token(pyramid_request)
-
-        assert result is None
-
-    def test_returns_none_for_empty_token(self, pyramid_request):
-        pyramid_request.headers["Authorization"] = "Bearer "
-
-        result = tokens.auth_token(pyramid_request)
-
-        assert result is None
-
     @pytest.mark.parametrize(
-        "header",
-        [
-            "",
-            "abcdef123",
-            "\x10",
-            ".\x00\"Ħ(\x12'𨳂\x05\U000df02a\U00095c2c셀",
-            "\U000f022b\t\x07\x1c0\x04\x06",
-        ],
+        "header,expected",
+        (
+            ("Bearer abcdef123", "abcdef123"),
+            (None, None),
+            ("Bearer ", None),
+            ("", None),
+            ("abcdef123", None),
+            ("\x10", None),
+            (".\x00\"Ħ(\x12'𨳂\x05\U000df02a\U00095c2c셀", None),
+            ("\U000f022b\t\x07\x1c0\x04\x06", None),
+        ),
```

Comment:
Nice, sometimes this happens organically, as you find more cases after the original ones and you (I'm guilty of that) don't zoom out on the test file to realize you are testing the same thing with different outputs.